io-functions-pay-portal/GetPaymentInfo/handler.ts

```diff
@@ -1,39 +1,75 @@
 import * as express from "express";
-import * as t from "io-ts";
 
+import { Context } from "@azure/functions";
 import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+
+import { identity } from "fp-ts/lib/function";
+import { IApiClient } from "../clients/pagopa";
+
 import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
 import {
   withRequestMiddlewares,
   wrapRequestHandler
 } from "io-functions-commons/dist/src/utils/request_middleware";
 import {
-  IResponseErrorNotFound,
   IResponseSuccessJson,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
 
 import { PaymentRequestsGetResponse } from "../generated/definitions/PaymentRequestsGetResponse";
+import { withApiRequestWrapper } from "../utils/api";
+import { getLogger, ILogger } from "../utils/logging";
+import { ErrorResponses } from "../utils/responses";
+
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { RptIdFromString } from "italia-pagopa-commons/lib/pagopa";
+
+type IGetPaymentInfoHandler = (
+  context: Context,
+  rptId: RptIdFromString
+) => Promise<IResponseSuccessJson<PaymentRequestsGetResponse> | ErrorResponses>;
 
-type IHttpHandler = () => Promise<
-  IResponseSuccessJson<PaymentRequestsGetResponse> | IResponseErrorNotFound
->;
+const logPrefix = "GetPaymentInfoHandler";
 
-export function HttpHandler(): IHttpHandler {
-  return async () => {
-    return ResponseSuccessJson({
-      codiceContestoPagamento: "testcode",
-      importoSingoloVersamento: 101
-    } as PaymentRequestsGetResponse);
+const getPaymentInfoTask = (
+  logger: ILogger,
+  apiClient: IApiClient,
+  rptId: RptIdFromString
+): TaskEither<ErrorResponses, PaymentRequestsGetResponse> =>
+  withApiRequestWrapper<PaymentRequestsGetResponse>(
+    logger, // tslint:disable-next-line: no-any
+    (): any =>
```

Comment:
thanks @infantesimone, I opted for the first one. Because in this way all three functions will have a common structure and it will be easier for those who have to make interventions later.